data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/configuration/MongoDBSourceConfig.java

```diff
@@ -57,7 +57,6 @@ public class MongoDBSourceConfig {
     private Boolean directConnection;
 
     @JsonProperty("aws")
-    @NotNull
```

Comment:
We should also do a migration to use the AwsCredentialsSupplier in this plugin so that the default role is used if not provided. But this can come in a different PR

Reply:
Yes, that is the plan